### PR TITLE
Handling initial paging requests over a message history of size one

### PR DESCRIPTION
**Pull request by staheri14**
Closes https://github.com/status-im/nim-waku/issues/471
## Problem

The pagination fails to handle initial paging requests when the store node holds only one message. See details in https://github.com/status-im/nim-waku/issues/471. 

In this PR, further test scenarios are developed to cover the identified corner cases.

### CODE REVIEW

**Review by D4nte — APPROVED** [review]
I haven't played with pagination yet but I noticed the bug.
LGTM thanks to the added tests.
I Haven't tested

File: waku/v2/protocol/waku_store/waku_store.nim [hunk 1]
```diff
@@ -254,15 +254,18 @@ proc paginateWithIndex*(list: seq[IndexedWakuMessage], pinfo: PagingInfo): (seq[
   var newCursor: Index # to be returned as part of the new paging info
   case dir
     of PagingDirection.FORWARD: # forward pagination
-      let remainingMessages= uint64(msgList.len) - uint64(foundIndex) - 1
+      # the message that is pointed by the cursor is excluded for the retrieved list, this is because this message has already been retrieved by the querier in its prior request
+      var remainingMessages= uint64(msgList.len) - uint64(foundIndex) - 1 
+      if initQuery:  remainingMessages = remainingMessages + 1
```
Comment (oskarth) [inline waku/v2/protocol/waku_store/waku_store.nim:259]: I wonder if this logic can't be simplified...minor thing though, and if tests are improved now that's great!

**Review by oskarth — APPROVED** [review]

**Review by jm-clius — APPROVED** [review]
lgtm :)

**Review by richard-ramos — APPROVED** [review]

File: waku/v2/protocol/waku_store/waku_store.nim (hunk 1 above)
Comment (staheri14) [reply waku/v2/protocol/waku_store/waku_store.nim:259]: I agree, was planing to do so in the near future. Here is issue I just opened for this sake https://github.com/status-im/nim-waku/issues/484
